### PR TITLE
GH-1882: Added Alias EnsureDirectoryDoesNotExist

### PR DESCRIPTION
**Pull request by nikhilagrawaldotnet**
@gep13 @augustoproiete @devlead - For #1882

### CODE REVIEW

File: src/Cake.Common/IO/DirectoryAliases.cs [hunk 1]
```diff
@@ -319,6 +320,49 @@ public static void EnsureDirectoryExists(this ICakeContext context, DirectoryPat
...
+        /// <example>
+        /// <code>
+        /// EnsureDirectoryDoNotExists("./be/gone");
+        /// </code>
+        /// </example>
+        /// <param name="context">The context.</param>
+        /// <param name="path">The directory path.</param>
+        [CakeMethodAlias]
+        [CakeAliasCategory("DoNotExists")]
+        public static void EnsureDirectoryDoNotExists(this ICakeContext context, DirectoryPath path)
```
Comment (augustoproiete) [inline src/Cake.Common/IO/DirectoryAliases.cs:335]: I believe we should name it `EnsureDirectoryDoesNotExist` to be grammatically correct

```suggestion
        public static void EnsureDirectoryDoesNotExist(this ICakeContext context, DirectoryPath path)
```

File: src/Cake.Common/IO/DirectoryAliases.cs [hunk 2]
```diff
@@ -319,6 +320,49 @@ public static void EnsureDirectoryExists(this ICakeContext context, DirectoryPat
...
+        ///     Force = true
+        /// });
+        /// </code>
+        /// </example>
+        /// <param name="context">The context.</param>
+        /// <param name="path">The directory path.</param>
+        /// <param name="settings">The delete settings.</param>
+        [CakeMethodAlias]
+        [CakeAliasCategory("DoNotExists")]
+        public static void EnsureDirectoryDoNotExists(this ICakeContext context, DirectoryPath path, DeleteDirectorySettings settings)
```
Comment (augustoproiete) [inline src/Cake.Common/IO/DirectoryAliases.cs:356]: Same as above... We should name it `EnsureDirectoryDoesNotExist` to be grammatically correct

```suggestion
        public static void EnsureDirectoryDoesNotExist(this ICakeContext context, DirectoryPath path, DeleteDirectorySettings settings)
```

Comment (augustoproiete) [reply src/Cake.Common/IO/DirectoryAliases.cs:356]: `DeleteDirectorySettings` makes a lot of sense to me, but I'm thinking if we shouldn't (for consistency) add an overload that receives an `EnsureDirectoryDoesNotExistSettings` (new, to be added to this PR, inheriting from `DeleteDirectorySettings`) which is what devs used to Cake would expect.

We could keep this overload with `DeleteDirectorySettings` as well as it's a good.

```csharp
EnsureDirectoryDoesNotExist(DirectoryPath)
EnsureDirectoryDoesNotExist(DirectoryPath, DeleteDirectorySettings)
EnsureDirectoryDoesNotExist(DirectoryPath, EnsureDirectoryDoesNotExistSettings)
```

@cake-build/cake-team & @nikhilagrawaldotnet what do you think?

**Review by augustoproiete — CHANGES_REQUESTED** [review]
Concept LGTM. Added a couple of comments for discussion.

File: src/Cake.Common/IO/DirectoryAliases.cs (hunk 1 above)
Comment (nikhilagrawaldotnet) [reply src/Cake.Common/IO/DirectoryAliases.cs:335]: Totally on board with `DoesNot`. 

`Exists` is used for consistency with `EnsureDirectoryExists`. What do you suggest?

Comment (augustoproiete) [reply src/Cake.Common/IO/DirectoryAliases.cs:335]: They both look grammatically correct to me - though English is not my main language, so correct if I'm wrong:

- `EnsureDirectoryExists` - Ensure it exists (plural exists)
- `EnsureDirectoryDoesNotExist` - Ensure it does not exist (singular exist)

File: src/Cake.Common/IO/DirectoryAliases.cs (hunk 2 above)
Comment (augustoproiete) [reply src/Cake.Common/IO/DirectoryAliases.cs]: @nikhilagrawaldotnet Let's go with adding the additional overload `EnsureDirectoryDoesNotExistSettings` if you don't mind and get this merged in. Everything else looks great as-is.

Comment (nikhilagrawaldotnet) [reply src/Cake.Common/IO/DirectoryAliases.cs]: @gep13 @devlead - FYI ^

Comment (augustoproiete) [reply src/Cake.Common/IO/DirectoryAliases.cs]: @nikhilagrawaldotnet Please ping us when you update the PR

**Review by augustoproiete — CHANGES_REQUESTED** [review]
https://github.com/cake-build/cake/pull/3409#discussion_r675846831

**Comment by augustoproiete** [comment]
Hey @nikhilagrawaldotnet yes :arrow_right: https://github.com/cake-build/cake/pull/3409#discussion_r674161607

**Comment by nikhilagrawaldotnet** [comment]
@augustoproiete - Change has been committed. Could you review, approve and merge pls.

**Review by nils-a — APPROVED** [review]
LGTM!

**Review by augustoproiete — APPROVED** [review]
LGTM

**Comment by augustoproiete** [comment]
@nikhilagrawaldotnet your changes have been merged, thanks for your contribution :+1:

**Pull request merged**